File: C/0242-Valid-Anagram.cpp
    bool isAnagram(string s, string t) {
        // Method 1
        sort(s.begin(), s.end());
        sort(t.begin(), t.end());
        return s == t;
        
        // Method 2
        if (s.length() != t.length()) {
            return false;
        } else {
            vector<int> letters(26, 0);
            
            for (int i = 0; i < s.length(); ++i) {
                letters[s[i]-'a']++;
                letters[t[i]-'a']--;
            }
            
            return letters == vector<int>(26, 0);
        }
    }

File: Python/0242-Valid-Anagram.py
    def isAnagram(self, s, t):
        // Method 1
        return sorted(s) == sorted(t)
        
        // Method 2
        if len(s) != len(t):
            return False
        d = dict((letter, 0) for letter in string.ascii_lowercase)
        
        for (a, b) in zip(s, t):
            d[a] += 1
            d[b] -= 1

        return not reduce(lambda x,y: x or y, d.values(), False)
